README.md

```diff
@@ -1,3 +1,125 @@
-# Utopia Logger library
+# Utopia Logger
 
-Work in progress. You can see development version in [dev branch](https://github.com/utopia-php/logger/tree/dev).
+[![Build Status](https://travis-ci.org/utopia-php/logger.svg?branch=main)](https://travis-ci.com/utopia-php/logger)
+![Total Downloads](https://img.shields.io/packagist/dt/utopia-php/logger.svg)
+[![Discord](https://img.shields.io/discord/564160730845151244)](https://appwrite.io/discord)
+
+Utopia Logger library is simple and lite library for logging information, such as errors or warnings. This library aims to be as simple and easy to learn and use as possible. This library is maintained by the [Appwrite team](https://appwrite.io).
+
+Although the library was built for the [Utopia Framework](https://github.com/utopia-php/framework) project, it is completely independent, **dependency-free** and can be used with any other PHP project or framework.
+
+## Getting Started
+
+Install using composer:
+```bash
+composer require utopia-php/logger
+```
+
+```php
+<?php
+
+require_once '../vendor/autoload.php';
+
+use Utopia\Logger\Adapter\AppSignal;
+use Utopia\Logger\Adapter\Raygun;
+use Utopia\Logger\Adapter\Sentry;
+use Utopia\Logger\Log;
+use Utopia\Logger\Log\Breadcrumb;
+use Utopia\Logger\Log\User;
+use Utopia\Logger\Logger;
+
+// Prepare log
+$log = new Log();
+$log->setAction("controller.database.deleteDocument");
+$log->setEnvironment("production");
+$log->setNamespace("api");
+$log->setServer("digitalocean-us-001");
+$log->setType(Log::TYPE_WARNING);
+$log->setVersion("0.11.5");
+$log->setMessage("Document efgh5678 not found");
+$log->setUser(new User("efgh5678"));
+$log->addBreadcrumb(new Breadcrumb(Log::TYPE_DEBUG, "http", "DELETE /api/v1/database/abcd1234/efgh5678", \microtime(true) - 500));
+$log->addBreadcrumb(new Breadcrumb(Log::TYPE_DEBUG, "auth", "Using API key", \microtime(true) - 400));
+$log->addBreadcrumb(new Breadcrumb(Log::TYPE_INFO, "auth", "Authenticated with * Using API Key", \microtime(true) - 350));
+$log->addBreadcrumb(new Breadcrumb(Log::TYPE_INFO, "database", "Found collection abcd1234", \microtime(true) - 300));
+$log->addBreadcrumb(new Breadcrumb(Log::TYPE_DEBUG, "database", "Permission for collection abcd1234 met", \microtime(true) - 200));
+$log->addBreadcrumb(new Breadcrumb(Log::TYPE_ERROR, "database", "Missing document when searching by ID!", \microtime(true)));
+$log->addTag('sdk', 'Flutter');
+$log->addTag('sdkVersion', '0.0.1');
+$log->addTag('authMode', 'default');
+$log->addTag('authMethod', 'cookie');
+$log->addTag('authProvider', 'MagicLink');
+$log->addExtra('urgent', false);
```

Comment:
Can we think of a better name than Extra? Seems very generic and not intuitive. Why can't we use `addTag()` for this data as well?

Reply:
That's probably only because `tags` and `extras` are displayed differently in the UIs of these adapters.

Reply:
As @christyjacob4 mentioned, some APIs supports them and it makes sense to me. Tags are like a label so you can easily filter, sort, search... Extra is more like an JSON object with bunch of additional data.

For example, `version` is great example of `tag`, because you want to filter by this information.
`contactFormSubject` is great example of `extra`, because it might help you debug the error, but you will never search by **specific** text user wrote.

In other words, `tags` is there for finding a specific error, `extra` is there for understanding the error.

Reply:
Functionality wise, I'm happy with it, just the name feels not 100%. We can move on with this and re-iterate when we have a better idea.